internal/evaluator/conftest_evaluator.go

```diff
@@ -622,3 +657,46 @@ func ExtractStringFromMetadata(result output.Result, key string) string {
 	}
 	return ""
 }
+
+func withCapabilities(ctx context.Context, capabilities string) context.Context {
+	return context.WithValue(ctx, capabilitiesKey, capabilities)
+}
+
+// capabilities returns a JSON serialized OPA Capability meant to isolate rego
+// policies from accessing external information, such as hosts or environment
+// variables. If the context already contains the capability, then that is
+// returned as is. Use withCapabilities to pre-populate the context if needed.
+func capabilities(ctx context.Context) (string, error) {
+	if c, ok := ctx.Value(capabilitiesKey).(string); ok && c != "" {
+		return c, nil
+	}
+
+	capabilities := ast.CapabilitiesForThisVersion()
+	// An empty list means no hosts can be reached. However, a nil value means all
+	// hosts can be reached. Unfortunately, the required JSON marshalling process
+	// drops the "allow_net" attribute if it's an empty list. So when it's loaded
+	// by OPA, it's seen as a nil value. As a workaround, we add an empty string
+	// to the list which shouldn't match any host but preserves the list after the
+	// JSON dance.
+	capabilities.AllowNet = []string{""}
+
+	builtins := make([]*ast.Builtin, 0, len(capabilities.Builtins))
+	for _, b := range capabilities.Builtins {
+		switch b.Name {
+		case "opa.runtime":
```

Comment:
Do we want to log skipped capabilities? I'm just thinking if a user sees that the functions are undefined in stderr, maybe having it logged that it's not allowed may help.

Reply:
Good call. I had to restructure these lines to avoid having to repeat the function names, but it does log what is being skipped now. Thanks!